relay/relay.go

```diff
@@ -67,6 +67,18 @@ type CallFrame interface {
 	Arg2Append(key, val []byte)
 }
 
+// RespFrame is an interface that abstracts access to the CallRes frame
+type RespFrame interface {
+	// OK indicates whether the call was successful
+	OK() bool
+
+	// Arg2 returns the raw arg2 payload
+	Arg2() []byte
```

Comment:
Do we want to expose arg2?

Reply:
this is primarily for testing since the `thrift` package imports `tchannel`, and we can't really test arg2 iteration with the raw scheme. an alternative is to do a type assertion and inspect `lazyCallRes` directly, but even that would require exporting arg2 somehow. is there any other way we can test without exporting an arg2 accessor?

Reply:
ahh, I see. What are we doing for testing arg2 on the call req, can we do the same?

One potential follow up to maybe simplify things, we could change both the lazyCallReq and lazyCallRes to only export `Arg2() []byte` and `ArgScheme() []byte`, and then have the caller check the scheme, and if it's thrift, they call `NewKeyValueIterator` on the `Arg2() []byte` directly,
https://pkg.go.dev/github.com/uber/tchannel-go@v1.21.2/thrift/arg2#NewKeyValIterator

Reply:
we do export an accessor for the arg2 offset in the callReq, but it really doesn't feel any better than just exporting the slice directly. I do think though that exposing `Arg2()` and `Scheme()` is a good idea. I'll add a TODO for that.

Reply:
Added `Scheme()` to `RespFrame`